Feat/ignore hooks flag

DJ-Glock opened the pull request:
## Proposed changes
This PR contains changes to address the issue https://github.com/travisjeffery/mocha-teamcity-reporter/issues/56

- New reporter option was added to skip reporting for hooks with title containing some word.
- Also there is a minor fix of existing failing test for modern mocha versions
- Increased NodeJS version for Travis-CI (from 6 to 10)
- Removed prepare script from package.json because it caused an error while running tests in Travis-CI

## Types of changes

What types of changes does your code introduce
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

jamie-sherriff commented:
I see it has failed node 6 compatibility which I'm fine with as its long since end of life but will need to do  a major version bump for release.

jamie-sherriff commented:
@DJ-Glock  If you could please have a check over the Teamcity build on your branch just to make sure everything is coming through as you expect it to, there appears to be 42 new failing tests compared to the expected 7: https://teamcity.jetbrains.com/viewLog.html?buildId=3412162&tab=buildResultsDiv&buildTypeId=TeamCityThirdPartyPlugins_MochaTeamcityReporter_Build&branch_TeamCityThirdPartyPlugins_MochaTeamcityReporter=refs%2Fpull%2F57%2Fhead

DJ-Glock commented:
@jamie-sherriff I'll check.

DJ-Glock commented:
Hey @jamie-sherriff 
I've figured our why so many tests failed. I wrote tests for mocha 8.* that supports global hooks, but TC runs tests with mocha 6. So I've excluded tests for mocha 8 - moved them to separate folder. So in the future they can be used for mocha 8, now I do not know how to run only these tests using mocha 8.
![image](https://user-images.githubusercontent.com/26511191/114026717-3b69f380-987f-11eb-9a3e-e1a13e557f21.png)


I added tests that cover new functionality and work fine with mocha 6.

So there are 4 failing tests that should fail - tests from example suite. So in total we should expect 11 failed tests instead of 7.
https://teamcity.jetbrains.com/viewType.html?buildTypeId=TeamCityThirdPartyPlugins_MochaTeamcityReporter_Build&branch_TeamCityThirdPartyPlugins_MochaTeamcityReporter=refs%2Fpull%2F57%2Fhead&tab=buildTypeStatusDiv

DJ-Glock commented:
Note: in the latest push I just fixed missed semicolon.

DJ-Glock commented:
@jamie-sherriff 
Please review my latest changes:
- Increased NodeJS version for Travis-CI (from 6 to 10). As per [this document](https://nodejs.org/en/about/releases/) 10 version is in maintenance mode.  
- Removed prepare script from package.json because it caused an error while running tests in Travis-CI

I am not really familiar with npm prepare script and peer dependencies. But it did not work with prepare script that tried to run tests because mocha was not installed at the moment when prepare script was launched. So I removed it and travis-ci was able to run all tests. So now all checks are green.

TeamCity is also good, as I said before [TC detects 4 more failing tests](https://teamcity.jetbrains.com/viewType.html?buildTypeId=TeamCityThirdPartyPlugins_MochaTeamcityReporter_Build&branch_TeamCityThirdPartyPlugins_MochaTeamcityReporter=refs%2Fpull%2F57%2Fhead&tab=buildTypeStatusDiv), which is expected behaviour.

So I've updated PR description and after your review it should be ready for merge.

jamie-sherriff commented:
Thanks @DJ-Glock, sorry have been a bit occupied lately but will go over this in the next few days, looks good so far.

jamie-sherriff commented on package.json:
```diff
@@ -14,8 +14,7 @@
     "test-teamcity-example": "mocha test/example/exampleSimple.js --reporter lib/teamcity.js",
     "lint": "eslint lib/*.js test/**/*.js",
-    "lint-fix": "eslint --fix lib/*.js test/**/*.js",
-    "prepare": "npm run test"
```
What's the reason behind dropping prepare? However I didn't realize it was in so many npm lifecycle events now.

probably prepack would be more appropriate for context it was being used to run the tests before being packaged and published to npm. I don't mind tidying that up once your PR is merged in.

DJ-Glock replied:
I removed prepare script because travis-ci failed to execute it. It tried to run test script and failed because no mocha is installed at the moment when prepare script is launched. My guess is: it happened because mocha is not part of dev dependency. Not sure if it should be installed because of peer dependency.

DJ-Glock replied:
proof: https://travis-ci.org/github/travisjeffery/mocha-teamcity-reporter/builds/766902127

DJ-Glock replied:
So do you expect anything else from my side?
I tried to put `npm install mocha & npm run test` into prepare script, but it did not work https://travis-ci.org/github/travisjeffery/mocha-teamcity-reporter/builds/766904317

jamie-sherriff replied:
Nah, ill investigate that thanks for solving it in the meantime!

jamie-sherriff commented:
Looks great and thanks!

I will merge my pending changes in as well and do a release in the coming days.
